File: src/JavaScript/easy/array/problem1431.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    let greaterCandy = Math.max(...candies)
    let res = []
    candies.forEach(candy => {
        (extraCandies + candy >= greaterCandy) ? res.push(true) : res.push(false)
    })
    return res
};


// let candies = [2,3,5,1,3], extraCandies = 3
// let candies = [4,2,1,1,2], extraCandies = 1
let candies = [12,1,12], extraCandies = 10

kidsWithCandies(candies, extraCandies)

File: src/Python/easy/array/problem1431.py
from typing import List

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        greaterCandy = max(candies)
        res = []
        for candy in candies:
            if candy + extraCandies >= greaterCandy:
                res.append(True)
            else:
                res.append(False)
        return res

candies = [2,3,5,1,3]; extraCandies = 3

solution = Solution()
solution.kidsWithCandies(candies, extraCandies)